src/Umbraco.Forms.Integrations.Crm.Hubspot/App_Plugins/UmbracoForms.Integrations/Crm/Hubspot/hubspotfields.component.js

```diff
@@ -56,7 +56,7 @@ function HubSpotFieldsController($scope, $routeParams, umbracoFormsIntegrationsC
         if (!vm.setting.value) {
             vm.mappings = [];
         } else {
-            vm.mappings = JSON.parse(vm.setting.value);
+            vm.getParsedMappings();
```

Comment:
Probably this shouldn't be a method on the view model, rather can be a "private" function.  And naming seems a little off given it has a "get" prefix but doesn't actually return anything. 

Perhaps?

```
  vm.mappings = parseMappings(vm.setting.value);

function parseMappings(settingValue) {
  var mappings = JSON.parse(settingValue);
  return mappings.map(mappingObj => {
      if (mappingObj.appendValue === undefined) {
          mappingObj.appendValue = false;
      }
      return mappingObj;
  });
}
```

Reply:
Updated